test/SILOptimizer/noimplicitcopy_borrowing_parameters.swift

```diff
@@ -0,0 +1,574 @@
+func borrowValDefault<T : P>(_ x: GenericNonTrivialStruct<T>) {}
+func borrowValBorrowing<T : P>(_ x: borrowing GenericNonTrivialStruct<T>) {}
+func consumeVal(_ x: consuming NonTrivialStruct) {}
+func consumeVal<T : P>(_ x: consuming GenericNonTrivialStruct<T>) {}
+
+///////////////////////////////////////////
+// MARK: Simple Loadable Borrowing Tests //
+///////////////////////////////////////////
+
+func testLoadableBorrowingConsume(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    let _ = x // expected-note {{consumed here}}
+    let _ = x.k
+}
+
+func testLoadableBorrowingConsume2(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    var y = x // expected-note {{consumed here}}
+    y = x // expected-note {{consumed here}}
+    _ = y
+}
+
+func testLoadableBorrowingConsume3(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    let y = x // expected-note {{consumed here}}
+    _ = y
+}
+
+func testLoadableBorrowingUse(_ x: borrowing NonTrivialStruct) {
+    _ = x
+}
+
+func testLoadableBorrowingUseAndConsume(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    _ = x
+    let _ = x // expected-note {{consumed here}}
+}
+
+func testLoadableBorrowingCallBorrowValDefault(_ x: borrowing NonTrivialStruct) {
+    borrowValDefault(x)
+}
+
+func testLoadableBorrowingCallBorrowValBorrowing(_ x: borrowing NonTrivialStruct) {
+    borrowValBorrowing(x)
+}
+
+func testLoadableBorrowingCallMethodSelfDefault(_ x: borrowing NonTrivialStruct) {
+    x.doSomethingDefault()
+}
+
+func testLoadableBorrowingCallMethodSelfBorrowing(_ x: borrowing NonTrivialStruct) {
+    x.doSomethingBorrowing()
+}
+
+func testLoadableBorrowingEscapingClosure(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' cannot be captured by an escaping closure since it is a borrowed parameter}}
+    var f: () -> () = {}
+    f = { // expected-note {{closure capturing 'x' here}}
+        _ = x
+    }
+    _ = f
+}
+
+func testLoadableBorrowingNonEscapingClosure(_ x: borrowing NonTrivialStruct) {
+    func useNonEscaping(_ f: () -> ()) {}
+    useNonEscaping {
+        _ = x
+    }
+}
+
+func testLoadableBorrowingConsumeOperator(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    _ = consume x // expected-note {{consumed here}}
+}
+
+func testLoadableBorrowingEnum(_ x: borrowing LoadableEnum) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    switch x { // expected-note {{consumed here}}
+    case let .x(y):
+        _ = y
+        break
+    case .y:
+        break
+    }
+}
+
+func testLoadableBorrowingCopyOperator(_ x: borrowing NonTrivialStruct) {
+    _ = copy x
+    let _ = copy x
+    consumeVal(copy x)
+}
```

Comment:
I think we should mix `copy x` into more situations to make sure its got coverage, since that's what makes it different from the noncopyable types. For example, we could take this switch test and write a positive / corrected version with `switch copy x` which I don't think we've tried to ensure it's fine with it.